sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -560,8 +560,10 @@ static pi_result copyModule(ze_context_handle_t ZeContext,
 
 static bool setEnvVar(const char *var, const char *value);
 
-static pi_result getOrCreatePlatform(ze_driver_handle_t ZeDriver,
-                                     pi_platform *Platform);
+static pi_result
+getPlatformCache(std::vector<pi_platform> const **PlatformCache);
```

Comment:
please avoid `**`, can't you use reference instead?

Reply:
I think it would be cleaner to not expose/return the "cache", just make an interface that updates the cache underneath.

Reply:
Personally, I think it's better to use a pointer when returning an "out" parameter, rather than using a reference.  The reason is that when reading the call site, it's obvious that the call will modify the parameter.  If you use reference, there's no way to know that the parameter will be modified unless you look at the function's declaration.

Reply:
> it's better to use a pointer when returning an "out" parameter, rather than using a reference

I totally agree to this. It should be enough to have a single pointer to do this, not a pointer to pointer.
And let's not have the cache be part of the interface at all.

Reply:
To clarify: I suggest not to return the entire platform's cache (and expose its structure as such) but just return the PI platform(s) from the cache: all of then for `piPltafromsGet` or just the one corresponding to a given optional L0 driver (new argument), The returned platforms would be of `pi_platform *` type then.

Reply:
I have refactored the code so that piPlatformsGet maintains the cache and piextPlatformCreateWithNativeHandle calls piPlatformsGet to query the cache.